### ISSUE TITLE
Code scan: Some HTML files are 99.9% javascript

### ISSUE THREAD

**Issue by Gagniuc** [github/linguist#5827]
## Describe the enhancement
I propose a code scan for HTML files. The .html extensions are cataloged as HTML, but javascript source code can be found inside the majority of these files in a large proportion, sometimes 99.9% is javascript (for instance my projects). Please see the html files from any of [these repositories](https://github.com/Gagniuc).

**Comment by Nixinova** [comment]
If its a .html file it'll be classified as HTML even if it has lots of stuff inside a <script> tag.

**Comment by Gagniuc** [comment]
The most simple solution may be:

a = html source code
b = replace "<script>" with nothing in a

if length of a <> length of b then the html file contains javascript

**Comment by Gagniuc** [comment]
> If its a .html file it'll be classified as HTML even if it has lots of stuff inside a <script> tag. Because its all HTML; the browser just decides that <script> elements should be treated differently.

The most simple solution may be a HTML/JS label:

a = html source code
b = replace "<script>" with nothing in a

if length of a <> length of b then the html file contains javascript

Let me give you another example. What if I have a file with a .php extension, but the content is 100% HTML? I am certain that you see where I am getting at. The same is with HTML files that contain alamoast 100% javascript source code.


PS: can you please help the cause of our VB6 community: [https://github.com/github/linguist/issues/5824](https://github.com/github/linguist/issues/5824)

**Comment by Nixinova** [comment]
> The same is with HTML files that contain alamoast 100% javascript source code.
> The most simple solution may be a HTML/JS label:
HTML/JS *is* just HTML though, having embedded JS is normal. But if its so much that the file is more JS than HTML thats what external .js files are for.

> What if I have a file with a .php extension, but the content is 100% HTML?
Then its PHP still. Having GitHub actually parse all the files trying to find embedded sublanguages would be way too resource intensive for something that needs to run on github a zillion times per second.

**Comment by lildude** [comment]
This topic has been brought up a few times in the past, for example, in the discussions around detecting Python in Jupyter files in https://github.com/github/linguist/issues/3316, R in Rmarkdown files in https://github.com/github/linguist/issues/5208, and even JavaScript in HTML files in https://github.com/github/linguist/discussions/5248 and more recently https://github.com/github/linguist/issues/5326 

In short, this is highly unlikely to happen. If you want your mostly JavaScript containing HTML files to be identified as JavaScript you can use an override.

**Comment by Gagniuc** [comment]
This will be solved one day. The point with HTML files and JS is that .html allows for one file applications with no dependencies. I believe the world already craves for that.
